### DISCUSSION TITLE
summerwind/actions-runner:latest

### DISCUSSION THREAD

**Discussion by saggl** [community/community#160684] [Actions] [answered]
is the docker file to the image somewhere findable? @summerwind 
best 
chris

https://github.com/actions/actions-runner-controller/blob/62eca94e45dce43bc52ec3b0cd3b59aff39147a4/charts/actions-runner-controller/values.yaml#L54

**Comment by BapAtRb** [comment]
Hey @summerwind,
Would it be possible for you to share the Dockerfile for actions-runner and actions-runner-controller?
We will highly appreciate this!
Thanks in advance!

**Comment by YorickH** [comment, marked as answer]
@saggl @BapAtRb The runner Dockerfiles can be found here: https://github.com/actions/actions-runner-controller/tree/master/runner and based on the Digest of the tags this is the actual latest: https://github.com/actions/actions-runner-controller/blob/master/runner/actions-runner.ubuntu-20.04.dockerfile

**Comment by SenthilC23** [comment]
@summerwind  If i want to install ansible, should i edit in this runner image? and and on top of this my Dockerfile should like CMD entrypoint.sh is this right?

**Reply by YorickH** [reply]
Just build your own image based on this one:
```
FROM summerwind/actions-runner:v2.320.0-ubuntu-22.04
RUN python3 -m pip install --user ansible
```

You could also use `:latest` as your tag but I advice against it as it might have unexpected results. 
You can find the available tags [here](https://hub.docker.com/r/summerwind/actions-runner/tags)

**Reply by younsl** [reply]
@SenthilC23 Sharing an additional custom use case for actions-runner package repository configuration.

Actions-runner uses archive.ubuntu.com and security.ubuntu.com declared in `/etc/apt/sources.list` as default package repositories. When these repositories go down, all pipelines with package installation fail. Here's a simple fix using regional mirrors:

Create additional-sources.list file:

```list
# Additional APT Sources
## Kakao Mirror
deb http://mirror.kakao.com/ubuntu/ jammy main restricted universe multiverse
deb http://mirror.kakao.com/ubuntu/ jammy-updates main restricted universe multiverse
deb http://mirror.kakao.com/ubuntu/ jammy-security main restricted universe multiverse
```

Dockerfile:

```dockerfile
FROM summerwind/actions-runner:v2.328.0-ubuntu-22.04
USER root

# Add additional APT sources for better availability
COPY additional-sources.list /tmp/
RUN cat /tmp/additional-sources.list >> /etc/apt/sources.list && \
    rm /tmp/additional-sources.list && \
    apt-get update

USER runner
```
This adds backup repositories without replacing the defaults, preventing build failures during outages.
